client/client.go

```diff
@@ -312,3 +312,51 @@ func readFile(path string) (string, int64, error) {
 	hash := hex.EncodeToString(hasher.Sum(nil))
 	return hash, bytes, nil
 }
+
+
+func RegisterRequest(agent string, src string, dst string) {
+	var source string
+	// resolve source agent name/alias and identify file to transfer
+	if strings.Contains(src, ":") {
+		arr := strings.Split(src, ":")
+		source = arr[len(arr)-1]
+	}
+	// parse the input
+	var lfn, block, dataset string
+	if strings.Contains(src, "#") { // it is a block name, e.g. /a/b/c#123
+		arr := strings.Split(src, "#")
+		dataset = arr[0]
+		block = arr[1]
```

Comment:
here you don't need to split, since block name is "/a/b/c#123" a whole string and not just block id part. Apart from that code looks good to me.